src/raw/mod.rs

```diff
@@ -783,22 +802,26 @@ struct StreamState<'f, S> {
     trans: usize,
     out: Output,
     aut_state: S,
+    done: bool,  // ('done' = true) means that there are no unexplored transitions in the current state. 
+                 // 'trans' value should be ignored when done is true.
 }
 
 impl<'f, A: Automaton> StreamWithState<'f, A> {
 
     fn new(fst: &'f FstMeta, data: &'f [u8], aut: A, min: Bound, max: Bound) -> Self {
-        let mut rdr = StreamWithState {
+        StreamWithState {
             fst,
             data,
             aut,
             inp: Vec::with_capacity(16),
```

Comment:
Less of a question for you, but we should explain the magic variable 16 somewhere.

Reply:
Will make it a constant.